mlfinlab/portfolio_optimization/mean_variance.py

```diff
@@ -112,41 +112,45 @@ def allocate(self, asset_names=None, asset_prices=None, expected_asset_returns=N
-                                  covariance_matrix=None, target_return=0.2, target_risk=0.01, risk_aversion=10):
-        # pylint: disable=eval-used, too-many-locals
+    def allocate_custom_objective(self, non_cvxpy_variables, cvxpy_variables, objective_function, constraints=None):
+        # pylint: disable=eval-used, exec-used
         """
         Create a portfolio using custom objective and constraints.
 
-        :param custom_objective: (dict) A custom objective function with custom constraints. You need to write it in the form
-                                        expected by cvxpy. The objective will be a single string while the constraints can be a
-                                        list of strings specifying the constraints. For e.g. {'objective': 'cp.Maximisie(
-                                        expected_asset_returns)', 'constraints': ['weights >= 0', 'weights <= 1']}.
-        :param asset_names: (list) A list of strings containing the asset names.
-        :param asset_prices: (pd.DataFrame) A dataframe of historical asset prices (daily close).
-        :param expected_asset_returns: (list/np.array/pd.dataframe) A list of mean stock returns (mu).
-        :param covariance_matrix: (pd.DataFrame/numpy matrix) User supplied covariance matrix of asset returns (sigma).
-        :param target_return: (float) Target return of the portfolio.
-        :param target_risk: (float) Target risk of the portfolio.
-        :param risk_aversion: (float) Quantifies the risk averse nature of the investor - a higher value means
-                                      more risk averse and vice-versa.
-        """
-
-        self._error_checks(asset_names, asset_prices, expected_asset_returns, covariance_matrix)
-
-        # Calculate the expected asset returns and covariance matrix if not given by the user
-        expected_asset_returns, cov = self._calculate_estimators(asset_prices,
-                                                                 expected_asset_returns,
-                                                                 covariance_matrix)
-
+        :param non_cvxpy_variables: (dict) A dictionary of variables to be used for providing the required input matrices and
+                                           other inputs required by the user. The key of dictionary will be the variable name
+                                           while the value can be anything ranging from a numpy matrix, list, dataframe or number.
+        :param cvxpy_variables: (list) This is a list of cvxpy specific variables that will be initialised in the format required
+                                       by cvxpy. For e.g. ["risk = cp.quad_form(weights, covariance)"] where you are initialising
+                                       a variable named "risk" using cvxpy. Note that cvxpy is being imported as "cp", so be sure
```

Comment:
Your not consistent, here you revert back to the double quotations. I feel like I am missing something?